apis/python/src/tiledbsoma/_dask/util.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) TileDB, Inc. and The Chan Zuckerberg Initiative Foundation
+#
+# Licensed under the MIT License.
+
+"""Utilities for loading a TileDB-SOMA ``X`` matrix as a Dask Array.
+"""
+
+from __future__ import annotations
+
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass, field
+from functools import cache
+from os import cpu_count
+from typing import (
+    Any,
+    Sequence,
+    Union,
+)
+
+import numpy as np
+import pyarrow as pa
+from numpy import int64
+from numpy.typing import NDArray
+from somacore.options import SparseNDCoord, SparseNDCoords
+from typing_extensions import TypeAlias
+
+from tiledbsoma.options import SOMATileDBContext
+from tiledbsoma.options._soma_tiledb_context import ConfigVal
+
+ChunkSize = Union[int, tuple[Union[int, None], int]]
+JoinIDs: TypeAlias = NDArray[int64]
+
+
+@dataclass
+class DaskConfig:
+    chunk_size: ChunkSize
+    tdb_concurrency: int | None = 1
```

Comment:
I think calling these `tiledb_concurrency` and `tiledb_configs` would be more lucid for the API, and would only 'cost' an additional three characters each.